### PR TITLE
Restructuring Guides IA and adding new sidebar to Docs

### PR DESCRIPTION
**Pull request by rachelnabors**
As per issue #1272, this restructuctures the guides into a sidebar. 

- Retains platform-specific sections
- Chunks guides out by topics of interest

? Maybe the sidebar should start expanded? But it's so much to show all at once!

### CODE REVIEW

File: website/sidebars.json [hunk 1]
```diff
@@ -115,7 +114,6 @@
       "image-style-props",
       "interactionmanager",
       "keyboard",
-      "layout-props",
```
Comment (Simek) [inline website/sidebars.json]: @rachelnabors Was `layout-props` page been removed intentionally?

Comment (rachelnabors) [reply website/sidebars.json]: That was unintentional! Good catch :)

**Comment by thymikee** [comment]
Hey, I admire your attempt at categorizing things, it's really hard to get right :D Looking at the new sidebar, I have a feeling that there's too many categories out there. I needed to open them one-by-one to see what's there, as some (like "Interoperability" or "JavaScript Internals") were not quite clear straight away. Thankfully we have an Algolia search for quick access, but still. I think we want to strive for as little categories as possible – that's my personal preference.

Some thoughts for possible improvements:
1. How about moving "Upgrading to new React Native versions" to "Workflow". The "Maintaining" category feels odd, and upgrading is a part of workflow, isn't it? 
2. How about renaming "JavaScript Internals" -> "JavaScript Runtime", since we're talking about JS engines and their runtime execution environments here?
3. Should we mix categories? E.g. "Using Hermes" fits into "Performance" as well, but I'm fine with current state.
4. "Platform Specific Code" fits into "Workflow" or "The Basics". Maybe there's no need for "Interoperability" category?

**Comment by rachelnabors** [comment]
@thymikee Thanks for your feedback! These are all good suggestions. Let me breakdown two important things you brought up:

> I have a feeling that there's too many categories out there. I needed to open them one-by-one to see what's there, as some (like "Interoperability" or "JavaScript Internals") were not quite clear straight away. Thankfully we have an Algolia search for quick access, but still. 

The reason for making categories is that metrics and interviews indicated that readers were not opening "Guides" to find things anyway: they were already using the search but more often heading to Google. We should aim, with these categories, to use words that help people with questions to find the general area of the content providing answers. In that regard, I'm eager for pushback on the category terminologies and happy to work with contributors on them even more!

> Should we mix categories?

Technically, there's no reason we can't put Hermes under both _JS Runtime_ and _Performance_! But is that the right thing to do? It could confuse users to have a sense of deja vu. Perhaps what we really need is a section on Hermes, or a page on "Hermes for performance" in the performance section.

I'll make your tweaks and push them up shortly. Thanks!

**Comment by thymikee** [comment]
I'm also not sold on mixing categories, was curious about your take on it. I think we're on the same page then.

**Pull request merged**